### PR TITLE
Restructure issue list template, styles

### PR DESCRIPTION
**Pull request by denyskon**
This PR does various modifications on the issue list shared template:
- restructure layout to achieve better responsiveness
-  fix various style issues
- restructure styles (better result with less code :)
- remove numerous `gt-*` patches and other unneeded classes -> use existing css classes

<details>
<summary>Before:</summary>

Bildschirmfoto vom 2023-07-07 14-35-00
Bildschirmfoto vom 2023-07-07 14-35-19
Bildschirmfoto vom 2023-07-07 14-35-43
</details>

<details>
<summary>After:</summary>

Bildschirmfoto vom 2023-07-07 14-32-04
Bildschirmfoto vom 2023-07-07 14-31-32
Bildschirmfoto vom 2023-07-07 14-31-14

</details>

### CODE REVIEW

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
One idea I had in mind is to move from `.issue.list` to a better name and a single class name. The styles originate from the issue list but a number of other lists now use it as well, so the name is not accurate any more. We'd need a somewhat generic name, maybe call it `flex-list` or similar. Doesn't have to be in this PR of course.

**Comment by denyskon** [comment]
@silverwind I plan to merge all the similar & often partially broken 'card styles/templates' into one at some point as most of them use a similar layout: leading icon/image, title, subtitle/body, trailing icons/buttons. I thought it would be better to start with this small refactoring.

I would prefer if we could merge this now, and an/some additional PR(s) will follow soon continuing with this approach.

**Comment by silverwind** [comment]
Yes, of course. Step by step.

File: web_src/css/shared/issuelist.css [hunk 1]
```diff
@@ -3,6 +3,30 @@
...
 
+.issue.list .item {
+  display: flex;
+  align-items: baseline;
+  padding: 8px 0;
+}
+
+.issue.list .item .issue-item-left {
+  display: flex;
+  align-items: start;
```
Comment (silverwind) [inline web_src/css/shared/issuelist.css]: Would `flex-start` also work? It has better browser support. Note, I don't understand the difference between them.

File: web_src/css/shared/issuelist.css [hunk 2]
```diff
@@ -54,108 +77,49 @@
   word-break: break-all;
 }
 
-.issue.list > .item .labels-list {
-  position: relative;
-  top: -1.5px;
+.issue.list > .item .title {
+  color: var(--color-text);
+  word-break: break-word;
```
Comment (silverwind) [inline web_src/css/shared/issuelist.css]:
```suggestion
  overflow-wrap: anywhere;
```

`word-break: break-word` is deprecated per [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break#break-word).

File: web_src/css/shared/issuelist.css (hunk 1 above)
Comment (denyskon) [reply web_src/css/shared/issuelist.css]: Should work...

Comment (silverwind) [reply web_src/css/shared/issuelist.css]: Then use it to better support browsers like Pale Moon which may not have it yet.

**Comment by denyskon** [comment]
I have started refactoring the other card elements to use this card structure, and it looks like it's easier to do it in one PR so nothing breaks in between...

**Review by wxiaoguang — APPROVED** [review]
It looks clearer than before, and it fixes more SVG alignments.

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Has conflicts.

**Comment by wxiaoguang** [comment]
The conflicts are caused by  Reformat some templates #25756 

That's what I said in https://github.com/go-gitea/gitea/pull/25739#discussion_r1255986318

> No reformat. To avoid unnecessary diff and unnecessary merge conflicts
> ...
> It's not a weak reason now since there are a lot of frontend refactoring.
> Otherwise some PRs would lose changes when resolving conflicts, for example, I fixed merge errors in `#25625`.....
> If there is no "indent" change, git can merge most of the changes flawlessly.

**Comment by silverwind** [comment]
Maybe if we could get the backend to just "minify" the insignificant whitespace, it would be ideal, but I'm not sure whether the extra the processing cost of such an effort is worth it.

**Comment by wxiaoguang** [comment]
To help merge this PR, you can do this:

1. Re-format the templates on your side following #25756
2.Then do a git merge, then git should be able to auto merge most changes without losing code

**Comment by silverwind** [comment]
Could probably do template optimization as a one-time action on process startup to optimize the templates and then cache the result.

**Comment by wxiaoguang** [comment]
I do not think it's worth to play with these spaces at the moment, neither the spaces between tags nor the spaces between class names.

Just like you won't look into the generated assemble code in a binary program, won't look into the generated JS code after packing, nobody would be hurt by the generated HTML code for browsers -- as long as the browser can render it correctly.

Code should be written in a developer-friendly style, but not in a generated-content-friendly style.

**Comment by silverwind** [comment]
Agree it's not worth from a review standpoint, but delivering minimal HTML is worth from performance standpoint.

**Comment by denyskon** [comment]
Fixed the conflicts

**Pull request merged**